src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  onUpdate: (id: number, updatedTodo: Partial<Todo>) => void;
+  onDelete: (id: number) => Promise<void>;
+  isTempToDo?: boolean;
```

Comment:
```suggestion
  isLoading: boolean;
```
And pass true for common todos during actions or for tempTodo

Reply:
not good